### PR TITLE
User can find song

### PR DESCRIPTION
**Pull request by dernathul**
https://www.pivotaltracker.com/story/show/171702430

```
As a User,
in order to find a song
i need to be able to be able to search for it in a form field
```
----------------------------------------------------------------
- Search by track added and working. 
- Added Cypress feature tests for searching tracks  
- Tests passing green locally and when receiving data from the API
- API responds with requested data
-----------------------------------------------------------------
We refreshed our knowledge about iterating over an Array, we also learned to work as a team
 and to ask  properly  scoped questions when asking for help

### CODE REVIEW

**Comment by dernathul** [comment]
Test finally passed locally, also test with the api.

**Comment by dernathul** [comment]
https://www.pivotaltracker.com/story/show/171702430

moving to create a search by artist

File: cypress/integration/userCanFindSong.feature.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+     cy.server();
+     cy.route({
+      method: 'GET',
+      url: "http://localhost:3000/api/v1/tracks**",
+      status: "400",
+      response: {
+        error_message: "There is no matches for the song you are trying to search"
+      }
+    })
+      cy.visit("/");
```
Comment (oliverochman) [inline cypress/integration/userCanFindSong.feature.js]: fix indentaion in this file

File: src/App.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import React, { Component } from "react";
+import axios from "axios";
+
+class App extends Component {
+  state = {
+    query: "",
+    error_message: {}
```
Comment (oliverochman) [inline src/App.jsx]: you dont need this line

File: src/App.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    return (
+
+      <>
+      <form onSubmit={this.onSubmitHandler}>
+        <input
+          id="search-field"
+          name="query"
+        />
+        <button type="submit" id="search">
+          {" "}
```
Comment (oliverochman) [inline src/App.jsx]: why do you need this line

File: src/App.jsx (hunk 2 above)
Comment (dernathul) [reply src/App.jsx]: ok, removed

File: cypress/integration/userCanFindSong.feature.js (hunk 1 above)
Comment (dernathul) [reply cypress/integration/userCanFindSong.feature.js]: yeah, we have formated everythjing

**Comment by dernathul** [comment]
Changes fixed, thank you for your comments

**Pull request merged**